src/libs/actions/IOU.js

```diff
@@ -682,9 +684,10 @@ function getMoneyRequestInformation(
         chatReport = ReportUtils.buildOptimisticChatReport([payerAccountID]);
     }
 
-    // STEP 2: Get existing IOU report and update its total OR build a new optimistic one
-    const isNewIOUReport = !chatReport.iouReportID || ReportUtils.hasIOUWaitingOnCurrentUserBankAccount(chatReport);
-    let iouReport = isNewIOUReport ? null : allReports[`${ONYXKEYS.COLLECTION.REPORT}${chatReport.iouReportID}`];
+    // STEP 2: Get existing IOU report, either from the chat report, or by the provided specific moneyRequestReportID,
+    // then update its total OR build a new optimistic one.
+    const isNewIOUReport = !moneyRequestReportID && (!chatReport.iouReportID || ReportUtils.hasIOUWaitingOnCurrentUserBankAccount(chatReport));
+    let iouReport = isNewIOUReport ? null : allReports[`${ONYXKEYS.COLLECTION.REPORT}${moneyRequestReportID > 0 ? moneyRequestReportID : chatReport.iouReportID}`];
```

Comment:
NAB - This seems a bit cleaner to me, what do you think? It could be too verbose, just thinking out loud :D

```suggestion
    let iouReport = null;
    const isNewIOUReport = !moneyRequestReportID && (!chatReport.iouReportID || ReportUtils.hasIOUWaitingOnCurrentUserBankAccount(chatReport));
    if (moneyRequestReportID > 0) {
        iouReport = allReports[`${ONYXKEYS.COLLECTION.REPORT}${moneyRequestReportID}`];
    } else if (!isNewIOUReport) {
        iouReport = allReports[`${ONYXKEYS.COLLECTION.REPORT}${chatReport.iouReportID}`];
    }
```

Reply:
Updated!

Reply:
@mountiny I'm working on [TS migration of IOU file](https://github.com/Expensify/App/pull/34092), and after merging the latest main I have a question regarding this part of the code.
Since `moneyRequestReportID` can be a `string`, TS complains on this check
```
if (moneyRequestReportID > 0) {
```
Was there any reason why you decided to have this specific check or may I update it just to be
```
if (moneyRequestReportID) {
```
Also for type consistency, I would update moneyRequestReportID default values form `0` to empty string `''` [here](https://github.com/Expensify/App/blob/main/src/libs/actions/IOU.js#L687), [here](https://github.com/Expensify/App/blob/main/src/libs/actions/IOU.js#L903) and [here](https://github.com/Expensify/App/blob/main/src/libs/actions/IOU.js#L1326), what do you think?

Reply:
@VickyStash Thank I think that makes sense to do now given the reportIDs are typed as string. In backend its integer so might be a bit confusing for internal devs